File: binding/include/charts/libqpielegendmarker.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qpielegendmarker.html
pub const qpielegendmarker = struct {

    /// New constructs a new QPieLegendMarker object.
    pub fn New(series: ?*anyopaque, slice: ?*anyopaque, legend: ?*anyopaque) QtC.QPieLegendMarker {
        return qtc.QPieLegendMarker_new(@ptrCast(series), @ptrCast(slice), @ptrCast(legend));
    }


    /// New2 constructs a new QPieLegendMarker object.
    pub fn New2(series: ?*anyopaque, slice: ?*anyopaque, legend: ?*anyopaque, parent: ?*anyopaque) QtC.QPieLegendMarker {
        return qtc.QPieLegendMarker_new2(@ptrCast(series), @ptrCast(slice), @ptrCast(legend), @ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QPieLegendMarker_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qpielegendmarker.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Type(self: ?*anyopaque, ) i32 {
        return qtc.QPieLegendMarker_Type(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnType(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) i32) void {
        qtc.QPieLegendMarker_OnType(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseType(self: ?*anyopaque, ) i32 {
        return qtc.QPieLegendMarker_QBaseType(@ptrCast(self));
    }

    pub fn Series(self: ?*anyopaque, ) QtC.QPieSeries {
        return qtc.QPieLegendMarker_Series(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnSeries(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) QtC.QPieSeries) void {
        qtc.QPieLegendMarker_OnSeries(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSeries(self: ?*anyopaque, ) QtC.QPieSeries {
        return qtc.QPieLegendMarker_QBaseSeries(@ptrCast(self));
    }

    pub fn Slice(self: ?*anyopaque, ) QtC.QPieSlice {
        return qtc.QPieLegendMarker_Slice(@ptrCast(self));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QPieLegendMarker_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qpielegendmarker.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QPieLegendMarker_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qpielegendmarker.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QPieLegendMarker_Delete(@ptrCast(self));
    }
};

File: binding/src/charts/libqpielegendmarker.cpp
#include <QLegend>
#include <QLegendMarker>
#include <QObject>
#include <QPieLegendMarker>
#include <QPieSeries>
#include <QPieSlice>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <qpielegendmarker.h>
#include "libqpielegendmarker.h"
#include "libqpielegendmarker.hxx"

QPieLegendMarker* QPieLegendMarker_new(QPieSeries* series, QPieSlice* slice, QLegend* legend) {
	 return new VirtualQPieLegendMarker(series, slice, legend);
}

QPieLegendMarker* QPieLegendMarker_new2(QPieSeries* series, QPieSlice* slice, QLegend* legend, QObject* parent) {
	 return new VirtualQPieLegendMarker(series, slice, legend, parent);
}

libqt_string QPieLegendMarker_Tr(const char* s) {
	QString _ret = QPieLegendMarker::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

int QPieLegendMarker_Type(QPieLegendMarker* self) {
	return self->type();
}

QPieSeries* QPieLegendMarker_Series(QPieLegendMarker* self) {
	return self->series();
}

QPieSlice* QPieLegendMarker_Slice(QPieLegendMarker* self) {
	return self->slice();
}

libqt_string QPieLegendMarker_Tr2(const char* s, const char* c) {
	QString _ret = QPieLegendMarker::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QPieLegendMarker_Tr3(const char* s, const char* c, int n) {
	QString _ret = QPieLegendMarker::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
int QPieLegendMarker_QBaseType(QPieLegendMarker* self) {
	auto* vqpielegendmarker = dynamic_cast<VirtualQPieLegendMarker*>(self);
	if (vqpielegendmarker && vqpielegendmarker->isVirtualQPieLegendMarker) {
vqpielegendmarker->setQPieLegendMarker_Type_IsBase(true);
	return vqpielegendmarker->type();
}
}

// Auxiliary method to allow providing re-implementation
void QPieLegendMarker_OnType(QPieLegendMarker* self, intptr_t slot) {
	auto* vqpielegendmarker = dynamic_cast<VirtualQPieLegendMarker*>(self);
	if (vqpielegendmarker && vqpielegendmarker->isVirtualQPieLegendMarker) {
vqpielegendmarker->setQPieLegendMarker_Type_Callback(reinterpret_cast<VirtualQPieLegendMarker::QPieLegendMarker_Type_Callback>(slot));
}
}

// Base class handler implementation
QPieSeries* QPieLegendMarker_QBaseSeries(QPieLegendMarker* self) {
	auto* vqpielegendmarker = dynamic_cast<VirtualQPieLegendMarker*>(self);
	if (vqpielegendmarker && vqpielegendmarker->isVirtualQPieLegendMarker) {
vqpielegendmarker->setQPieLegendMarker_Series_IsBase(true);
	return vqpielegendmarker->series();
}
}

// Auxiliary method to allow providing re-implementation
void QPieLegendMarker_OnSeries(QPieLegendMarker* self, intptr_t slot) {
	auto* vqpielegendmarker = dynamic_cast<VirtualQPieLegendMarker*>(self);
	if (vqpielegendmarker && vqpielegendmarker->isVirtualQPieLegendMarker) {
vqpielegendmarker->setQPieLegendMarker_Series_Callback(reinterpret_cast<VirtualQPieLegendMarker::QPieLegendMarker_Series_Callback>(slot));
}
}

void QPieLegendMarker_Delete(QPieLegendMarker* self) {
    delete self;
}

